guest/guest.go

```diff
@@ -66,3 +83,52 @@ func (d *Default) Apply(cluster *v1.Cluster) error {
 func (d Default) Delete(cluster *v1.Cluster) error {
 	panic("implement me")
 }
+
+func (d *Default) waitClusterReady(ctx context.Context) error {
+	var clusterStatusChan = make(chan string)
+	ctx, cancel := context.WithTimeout(ctx, 15*time.Minute)
+	defer cancel()
+	ticker := time.NewTicker(30 * time.Second)
+	defer ticker.Stop()
+	go func(t *time.Ticker) {
+		for {
+			clusterStatus := d.getClusterStatus()
+			clusterStatusChan <- clusterStatus
+			<-t.C
+		}
+	}(ticker)
+	for {
+		select {
+		case status := <-clusterStatusChan:
+			if status == common.ClusterNotReady {
+				logger.Info("wait cluster ready ")
+			} else if status == common.ClusterReady {
+				logger.Info("cluster ready now")
+				return nil
+			}
+		case <-ctx.Done():
+			return fmt.Errorf("cluster is not ready, please check")
+		}
+	}
+}
+
+func (d *Default) getClusterStatus() string {
+	k8sClient, err := k8s.Newk8sClient()
+	if err != nil {
+		logger.Warn(err)
```

Comment:
Shall we return when running into non-nil error?

Otherwise the following code would panic there since k8sClient is nil.